libs/ledger/src/chain/block_coordinator.cpp

```diff
@@ -1090,64 +1170,63 @@ BlockCoordinator::State BlockCoordinator::OnReset()
 
   FETCH_LOG_INFO(LOGGING_NAME, "Reset condition");
 
-  // trigger stake updates at the end of the block lifecycle
-  if (stake_ && beacon_)
+  Block const *block = nullptr;
+
+  if (next_block_)
   {
-    Block const *block = nullptr;
+    block = next_block_.get();
+  }
+  else if (current_block_)
+  {
+    block = current_block_.get();
+  }
+  else
+  {
+    FETCH_LOG_ERROR(LOGGING_NAME, "Unable to find a previously executed block!");
+  }
 
-    if (next_block_)
-    {
-      block = next_block_.get();
-    }
-    else if (current_block_)
-    {
-      block = current_block_.get();
-    }
-    else
-    {
-      FETCH_LOG_ERROR(LOGGING_NAME,
-                      "Unable to find a previously executed block when updating staking!");
-    }
+  // trigger stake updates at the end of the block lifecycle
+  if (stake_ && block)
+  {
+    stake_->UpdateCurrentBlock(*block);
+  }
 
+  if (beacon_)
+  {
     // If we are on the head of the main chain, and it is a 'committee triggering' block
-    if (block && (*chain_.GetHeaviestBlock() == *block))
+    // and we are in sync.
+    // TODO(HUT): this needs very carefully looking at!
+    if (block && (*chain_.GetHeaviestBlock() == *block) /*&& syncronised_*/)
     {
       auto const block_number = block->body.block_number;
 
-      if ((block_number % AEON_PERIOD) == 0)
+      if ((block_number % aeon_period_) == 0)
       {
-        std::unordered_set<fetch::crypto::Identity> cabinet_member_list;
-        auto                                        snapshot = (*stake_).GetCurrentStakeSnapshot();
-        auto const current_stakers = snapshot->BuildCommittee(0, MAX_COMMITTEE_SIZE);
+        beacon::BeaconService::CabinetMemberList cabinet_member_list;
+        auto const                               current_stakers = stake_->GetCommittee(*block);
 
-        // TODO(HUT): switch to set for stake manager
         for (auto const &staker : *current_stakers)
         {
           FETCH_LOG_DEBUG(LOGGING_NAME, "Adding staker: ", staker.identifier().ToBase64());
           cabinet_member_list.insert(staker);
         }
 
-        uint32_t threshold = uint32_t((cabinet_member_list.size() / 2) + 1);
+        // uint32_t threshold = uint32_t((cabinet_member_list.size() / 2) + 1);
+        uint32_t threshold = uint32_t(cabinet_member_list.size());
```

Comment:
That's a quite high threshold?

Reply:
100 node deployment did not work with a threshold less than 100%. I am working on it